main.go

```diff
@@ -70,34 +70,46 @@ func registerSIGINTHandler(mountPoint string) {
 	}()
 }
 
-func getUserAgent(appName string) string {
+func getUserAgent(appName string, configString string) string {
 	gcsfuseMetadataImageType := os.Getenv("GCSFUSE_METADATA_IMAGE_TYPE")
 	if len(gcsfuseMetadataImageType) > 0 {
-		userAgent := fmt.Sprintf("gcsfuse/%s %s (GPN:gcsfuse-%s)", getVersion(), appName, gcsfuseMetadataImageType)
+		userAgent := fmt.Sprintf("gcsfuse/%s %s (GPN:gcsfuse-%s) (Cfg:%s)", getVersion(), appName, gcsfuseMetadataImageType, configString)
 		return strings.Join(strings.Fields(userAgent), " ")
 	} else if len(appName) > 0 {
-		return fmt.Sprintf("gcsfuse/%s (GPN:gcsfuse-%s)", getVersion(), appName)
+		return fmt.Sprintf("gcsfuse/%s (GPN:gcsfuse-%s) (Cfg:%s)", getVersion(), appName, configString)
 	} else {
-		return fmt.Sprintf("gcsfuse/%s (GPN:gcsfuse)", getVersion())
+		return fmt.Sprintf("gcsfuse/%s (GPN:gcsfuse) (Cfg:%s)", getVersion(), configString)
 	}
 }
 
-func createStorageHandle(flags *flagStorage) (storageHandle storage.StorageHandle, err error) {
+func getConfigForUserAgent(mountConfig *config.MountConfig) string {
+	// Minimum configuration details created in a bitset fashion. Right now, its restricted only to File Cache Settings.
+	isFileCacheEnabled := "0"
+	if mountConfig.FileCacheConfig.MaxSizeMB != 0 && string(mountConfig.CacheDir) != "" {
+		isFileCacheEnabled = "1"
+	}
+	isFileCacheForRangeReadEnabled := "0"
+	if mountConfig.FileCacheConfig.CacheFileForRangeRead {
+		isFileCacheForRangeReadEnabled = "1"
+	}
+	return fmt.Sprintf("%s%s", isFileCacheEnabled, isFileCacheForRangeReadEnabled)
```

Comment:
add delimiter (:) between the config elements

Reply:
done